central/imageintegration/search/searcher_impl.go

```diff
@@ -0,0 +1,47 @@
+package search
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/central/imageintegration/index"
+	imageIntegrationMapping "github.com/stackrox/rox/central/imageintegration/index/mappings"
+	"github.com/stackrox/rox/central/imageintegration/store"
+	"github.com/stackrox/rox/central/role/resources"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/search/blevesearch"
+	"github.com/stackrox/rox/pkg/search/paginated"
+	"github.com/stackrox/rox/pkg/search/sortfields"
+)
+
+var (
+	defaultSortOption = &v1.QuerySortOption{
+		Field:    search.ClusterID.String(),
+		Reversed: false,
+	}
+	imageIntegrationSAC = sac.ForResource(resources.ImageIntegration)
+)
+
+// searcherImpl provides an intermediary implementation layer for image integration.
+type searcherImpl struct {
+	storage  store.Store
+	indexer  index.Indexer
+	searcher search.Searcher
+}
+
+func (ds *searcherImpl) Search(ctx context.Context, q *v1.Query) ([]search.Result, error) {
+	return ds.searcher.Search(ctx, q)
+}
+
+func (ds *searcherImpl) Count(ctx context.Context, q *v1.Query) (int, error) {
+	return ds.searcher.Count(ctx, q)
+}
+
+// Format the search functionality of the indexer to be filtered (for sac) and paginated.
+func formatSearcher(unsafeSearcher blevesearch.UnsafeSearcher) search.Searcher {
```

Comment:
@md2119 I don't think we need this for a new image integration searcher right?

Reply:
@md2119 bump ^